internal/cache/controller_obj_cache.go

```diff
@@ -2842,11 +2842,57 @@ func checkAndSetVRFFromNetwork(client *clients.AviClient) bool {
 		utils.AviLog.Infof("Using global VRF for NodePort mode")
 		return true
 	}
+	if lib.GetCloudType() == lib.CLOUD_NSXT && lib.GetServiceType() == "ClusterIP" && lib.GetCNIPlugin() != lib.NCP_CNI {
+		// Here we need to determine the right VRF for this T1LR
+		// The logic is: Get all the VRF context objects from the controller, figure out the VRF that matches the T1LR
+		// Current pagination size is set to 100, this may have to increased if we have more than 100 T1 routers.
+		uri := "/api/vrfcontext?" + "&include_name=true&cloud_ref.name=" + utils.CloudName + "&page_size=100"
```

Comment:
Should we have separate function to iterate through all items through pagination logic? (even if number of items more than 100, repeatedly calling same function with different page num should serve our purpose)

Reply:
100 VRFs would mean 100 T1s, I agree we should do pagination, as the comment reads - I have kept it for future. But I can include it in this PR too.

Reply:
Assuming 100 T1s, it won't be relevant unless there's a huge cluster. But programatically we should deal with it.

Reply:
We can take care of that in future PR. We can go ahead with this.